richtext-commonmark/src/androidMain/kotlin/com/halilibo/richtext/markdown/Actual.kt

```diff
@@ -0,0 +1,91 @@
+package com.halilibo.richtext.markdown
+
+import android.os.Build
+import android.text.Html
+import android.widget.TextView
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.BoxWithConstraints
+import androidx.compose.foundation.layout.size
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.isSpecified
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.viewinterop.AndroidView
+import coil.annotation.ExperimentalCoilApi
+import coil.compose.rememberImagePainter
+
+@Composable
+internal actual fun HtmlBlock(content: String) {
+  AndroidView(
+    factory = { context ->
+      // TODO: pass current styling to legacy TextView
+      TextView(context)
+    },
+    update = {
+      it.text = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+        Html.fromHtml(content, 0)
+      } else {
+        @Suppress("DEPRECATION")
+        Html.fromHtml(content)
+      }
+    }
+  )
+}
+
+/**
+ * Implementation of RemoteImage by using Coil library for Android.
+ */
+@OptIn(ExperimentalCoilApi::class)
+@Composable
+internal actual fun RemoteImage(
+  url: String,
+  contentDescription: String?,
+  modifier: Modifier,
+  contentScale: ContentScale
+) {
+  val painter = rememberImagePainter(data = url)
+
+  val density = LocalDensity.current
+
+  BoxWithConstraints {
+    val sizeModifier by remember(density, painter) {
+      derivedStateOf {
+        val painterIntrinsicSize = painter.state.painter?.intrinsicSize
+        if (painterIntrinsicSize != null &&
+          painterIntrinsicSize.isSpecified &&
+          painterIntrinsicSize.width != Float.POSITIVE_INFINITY &&
+          painterIntrinsicSize.height != Float.POSITIVE_INFINITY
+        ) {
+          val width = painterIntrinsicSize.width
+          val height = painterIntrinsicSize.height
+          val scale = if (width > constraints.maxWidth) {
+            constraints.maxWidth.toFloat() / width
+          } else {
+            1f
+          }
+
+          with(density) {
+            Modifier.size(
+              (width * scale).toDp(),
+              (height * scale).toDp()
+            )
+          }
+        } else {
+          Modifier.size(128.dp)
```

Comment:
Probably not in scope for this PR, but maybe worth a comment: How was this default size chosen? Should it be customizable?